File: src/main/java/com/hedvig/claims/commands/AddAutomaticPaymentCommand.java
package com.hedvig.claims.commands;

import javax.money.MonetaryAmount;
import lombok.Value;
import org.axonframework.commandhandling.TargetAggregateIdentifier;

@Value
public class AddAutomaticPaymentCommand {

  @TargetAggregateIdentifier
  String claimId;
  String memberId;
  MonetaryAmount amount;
  MonetaryAmount deductible;
  String note;
  boolean isExGracia;
  String handlerReference;
  boolean sanctionCheckSkipped;
}

File: src/main/java/com/hedvig/claims/commands/AddAutomaticPaymentCommand.kt
package com.hedvig.claims.commands

import com.hedvig.claims.query.Carrier
import org.axonframework.commandhandling.TargetAggregateIdentifier
import javax.money.MonetaryAmount

class AddAutomaticPaymentCommand(
    @TargetAggregateIdentifier
    val claimId: String,
    val memberId: String,
    val amount: MonetaryAmount,
    val deductible: MonetaryAmount,
    val note: String?,
    val isExGracia: Boolean,
    val handlerReference: String,
    val sanctionCheckSkipped: Boolean,
    val carrier: Carrier
)
